src/Sentry/Protocol/EnvelopeItem.cs

```diff
@@ -142,5 +146,24 @@ public static EnvelopeItem FromEvent(SentryEvent @event)
 
             return new EnvelopeItem(header, @event);
         }
+
+        public static async Task<EnvelopeItem> DeserializeAsync(
+            Stream stream,
+            CancellationToken cancellationToken = default)
+        {
+            // Header
+            var header = await Json.DeserializeFromStreamAsync<Dictionary<string, object>>(stream, cancellationToken)
+                .ConfigureAwait(false);
+
+            var length = (long)header.GetValueOrDefault(LengthKey, long.MaxValue);
```

Comment:
The `long.MaxValue` scares me. What's the use case/side effect for the default value?

Reply:
It's just a hack to get things working fast.